### ISSUE TITLE
Exercise 2

### ISSUE THREAD

**Issue by Duxon** [ContinuumIO/gtc2017-numba#2]
Hi,

I am struggling with exercise 2. My implementation looks like this:

```
def zero_suppress(waveform_value, threshold):
    for i in range(n)
        if waveform_value[i] < threshold: waveform_value[i] = 0
    return waveform_value
```

Somehow, that is not possible in CUDA and I don't understand why (if and for statements should be supported). Can someone elaborate?

**Comment by Duxon** [comment]
Okay, it seems like I was missing a fundamental concept in vectorization.
If I understand correctly, the function is executed n-times on a GPU device even though we call it with an array of dimension n. A correct implementation would, therefore, look like this:

```
if abs(waveform_value) > threshold:
    return waveform_value
else:
    return 0.0
```

**Comment by seibert** [comment]
Correct.  The `@vectorize` decorator handles the looping, which also means it generalizes to multidimensional arrays (not just 1D) as well.
